client/packages/inventory/src/Stocktake/ListView/CreateStocktakeModal.tsx

```diff
@@ -7,96 +7,132 @@ import {
   InputWithLabelRow,
   Typography,
 } from '@common/components';
-import { useFormatDateTime, useTranslation } from '@common/intl';
+import { DateUtils, useFormatDateTime, useTranslation } from '@common/intl';
 import { useDialog } from '@common/hooks';
 import {
-  useStockList,
+  useStockListCount,
   LocationSearchInput,
   LocationRowFragment,
   MasterListSearchInput,
   MasterListRowFragment,
 } from '@openmsupply-client/system';
-import { Box, Formatter } from '@openmsupply-client/common';
 import {
-  CreateStocktakeInput,
-  defaultCreateStocktakeInput,
-} from '../api/hooks/useStocktake';
+  Box,
+  Formatter,
+  StockLineFilterInput,
+  useNavigate,
+} from '@openmsupply-client/common';
+import { CreateStocktakeInput } from '../api/hooks/useStocktake';
 
 const LABEL_FLEX = '0 0 150px';
 interface NewStocktakeModalProps {
   open: boolean;
   onClose: () => void;
   onCreate: (input: CreateStocktakeInput) => Promise<string | undefined>;
   isCreating?: boolean;
-  navigate: (id: string) => void;
   description?: string;
 }
 
-// Intended behaviour is for the stocktake to generate based on one of the available argument selections only - the user cannot select multiple at once.
+interface ModalState {
+  location: LocationRowFragment | null;
+  masterList: MasterListRowFragment | null;
+  expiryDate: Date | null;
+  createBlankStocktake: boolean;
+}
 
 export const CreateStocktakeModal = ({
   open,
   onClose,
   onCreate,
   isCreating,
-  navigate,
   description,
 }: NewStocktakeModalProps) => {
+  const navigate = useNavigate();
```

Comment:
Rather than prop drilling.

Tempted for `isCreating,` `onCreate` too. `description` is tempting but it would create a tiny amount of duplication... which itself could be encapsulated in a hook rather than drillin' 😝